src/components/Code/Code.js

```diff
@@ -0,0 +1,150 @@
+
+  state = {
+    isCopying: false,
+  };
+
+  componentDidUpdate = () => {
+    // reselect text after re-render
+    if (this.state.selectedRange) {
+      const prevRange = this.state.selectedRange;
+      const sel = window.getSelection();
+      sel.removeAllRanges();
+      sel.addRange(prevRange);
+      // clear it for the next interaction
+      this.setState({ selectedRange: null });
+    }
+  };
+
+  handleCopyClick = () => {
+    // Create a textarea element in which the text of the code will be copied.
+    // Set the value of the textarea, select it to set it as the active element,
+    // then copy the value of that element to the clipboard.
+    // It's 2017 and this is still the best way to copy text on button click...
+    if (isFunction(this.props.onCopy)) {
+      this.props.onCopy();
+    }
+    const selection = document.getSelection();
+    const selectionString = selection.toString();
+    // if user has selected text, save that to state so it can be reselected
+    if (selectionString !== '') {
+      this.setState({
+        selectedRange: selection.getRangeAt(0),
+      });
+    }
+    // console.dir(document.getSelection().anchorNode);
+    const code =
+      selectionString === '' ? this.preNode.textContent : selectionString;
+    const txtArea = document.createElement('textarea');
+    // Safari doesn't allow for assigning an object literal to `style`.
+    txtArea.class = 'hidden-textarea';
+    // Make sure the code ends in a newline to execute all the commands in multiline code.
+    if (code.charAt(code.length - 1) === '\n') {
+      txtArea.value = code;
+    } else {
+      txtArea.value = `${code}\n`;
+    }
+    document.body.appendChild(txtArea);
+
+    txtArea.select();
+
+    try {
+      document.execCommand('copy');
+    } catch (e) {
+      throw e;
+    }
+    document.body.removeChild(txtArea);
+
+    // show the Copied to clipboard notice temporarily
+    this.setState({ isCopying: true });
+    setTimeout(() => {
+      this.setState({ isCopying: false });
+    }, 3000);
+  };
+
+  render() {
+    const { children } = this.props;
+    const { isCopying } = this.state;
+
+    const copyText = isCopying ? 'Copied to clipboard' : 'Copy to clipboard';
+    const copyIcon = isCopying ? 'files-o' : 'check';
+
+    return (
+      <CodeWrapper onClick={this.handleCopyClick}>
+        <Content>
+          <Pre innerRef={e => (this.preNode = e)}>
+            {isFunction(children) ? children() : trimString(children)}
+          </Pre>
+        </Content>
+
+        <CopyNotice>
+          <i className={`fa fa-${copyIcon}`} /> {copyText}
+        </CopyNotice>
+      </CodeWrapper>
+    );
+  }
+}
+
+Code.propTypes = {
+  // Children can be anything that React can render
+  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
+
+  // onCopy will be called when the CodeWrapper is clicked
+  onCopy: PropTypes.func,
+};
+
+Code.defaultProps = {
+  onCopy: () => {},
```

Comment:
There is `lodash.noop` which is maybe a tiny bit cleaner :)

Reply:
certainly is :+1: